src/Application.tsx

```diff
@@ -1,35 +1,59 @@
-import React, { Suspense } from "react";
+import React, { Suspense, useEffect } from "react";
 import {
   Route,
-  Switch,
   RouteComponentProps,
+  Switch,
+  withRouter,
 } from "react-router-dom";
-import routes from "src/common/routes";
 
-const Application: React.FunctionComponent<{}> = (props) => {
+import { guestRoutes, userRoutes } from "src/common/routes";
+import { routes } from "./common/constants/pageRoutes";
+import { useAuth } from "./context/GlobalContext";
+
+const Application: React.FC<
+  RouteComponentProps<{}, {}, { from: { pathname: string } }>
+> = ({ history }) => {
+  const { state } = useAuth();
+
+  useEffect(() => {
+    // Store previous route in history state to return user back to that route
+    // from which it came from after user verified from auth listener.
+    if (state.user) {
+      const { state } = history.location;
```

Comment:
Let's use more distinct variable names. Both the variable on line 16 and 22 have the same name.